projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -409,6 +409,10 @@ export class IgxFilteringService implements OnDestroy {
         if (expression.condition.isUnary) {
             return this.grid.resourceStrings[`igx_grid_filter_${expression.condition.name}`] || expression.condition.name;
         } else if (expression.searchVal instanceof Date) {
+            const formatter = this.grid.getColumnByName(expression.fieldName).formatter;
+            if (formatter) {
+                return formatter(expression.searchVal);
+            }
             const pipeArgs = this.grid.getColumnByName(expression.fieldName).pipeArgs;
```

Comment:
We could cache the column object.
```suggestion
            const column = this.grid.getColumnByName(expression.fieldName);
            const formatter = column.formatter;
            if (formatter) {
                return formatter(expression.searchVal);
            }
            const pipeArgs = column.pipeArgs;
```